op-alt-da/damgr_test.go

```diff
@@ -58,7 +58,7 @@ func TestFinalization(t *testing.T) {
 	state.Prune(bID(7))
 	require.Equal(t, eth.L1BlockRef{}, state.lastPrunedCommitment)
 	state.Prune(bID(8))
-	require.Equal(t, eth.L1BlockRef{}, state.lastPrunedCommitment)
+	require.Equal(t, l1Ref(bn1), state.lastPrunedCommitment)
```

Comment:
Out of curiosity do you know why did this test start failing?
Is op's circleci not running these tests? Prob should also change that as part of this PR?

Reply:
Also, isn't it this line that should be changed instead? https://github.com/ethereum-optimism/optimism/blob/f4506a8a77314cca8e27998dd28cf9a4f8b2cc90/op-alt-da/dastate.go#L101
to
```
challengeWindowEnd: inclusionBlock.Number + s.cfg.ChallengeWindow + 1,
```

The description for ChallengeWindow is https://github.com/ethereum-optimism/optimism/blob/f4506a8a77314cca8e27998dd28cf9a4f8b2cc90/op-alt-da/damgr.go#L56

So in this case given that the commitment was at block 2, and challengeWindow in the test is set to 6, then block 8 should still be challenageable, so it shouldn't have been prune (aka the initial require was correct, not your fix)

Reply:
Unfortunately tests were not being ran in the ci, I enable them here: https://github.com/ethereum-optimism/optimism/pull/14056, but I can combine both PRs if you prefer.

When I was going through the code it actually made sense to me that `resolveWindowEnd` was defined as it is, but I think you are right, will update. Thanks!

Reply:
Not my repo so its @tynes's call. But how do you plan on getting the other PR in if it doesn't pass CI?

Reply:
I was thinking that this could be merged first, but yeah will prob just merge that branch into this one

Reply:
merged, now this also enables tests in ci. Will now look into adapting the test. Thanks again.

Reply:
actually, @tynes could you confirm if the comment is incorrect or if I should actually fix the implementation? thanks

Reply:
Just ran CI, lets see if the tests pass. I dont have good insight into the safety of this change right now. Would it be a hardfork? Given that lack of certainty, I don't love the idea of merging that change. Going to need to rely on you and the other chains that run alt da in prod to ensure changes to altda are safe. I dont have the bandwidth to context switch into alt da and provide a lot of meaningful input on its technical details right now